sklego/meta/_grouped_utils.py

```diff
@@ -0,0 +1,62 @@
+from typing import Tuple
+
+import numpy as np
+import pandas as pd
+from scipy.sparse import issparse
+from sklearn.utils import check_array
+from sklearn.utils.validation import _ensure_no_complex_data
+
+from sklego.common import as_list
+
+
+def split_groups_and_values(
+    X, groups, name="", **kwargs
+) -> Tuple[pd.DataFrame, np.ndarray]:
+    general_checks(X, name=name)
+
+    try:
+        if isinstance(X, pd.DataFrame):
+            X_group = X.loc[:, as_list(groups)]
+            X_value = X.drop(columns=groups).values
+        else:
+            X_group = pd.DataFrame(X[:, as_list(groups)])
+            X_value = np.delete(X, as_list(groups), axis=1)
+    except (KeyError, IndexError):
+        raise ValueError(f"Could not drop groups {groups} from columns of X")
+
+    X_group = check_grouping_columns(X_group, **kwargs)
+    X_value = check_value_columns(X_value, **kwargs)
+
+    return X_group, X_value
+
+
+def general_checks(X, name):
+    _ensure_no_complex_data(X)
+
+    if issparse(X):  # sklearn.validation._ensure_sparse_format to complicated
+        raise ValueError(f"The estimator {name} does not work on sparse matrices")
+
+    if X.ndim == 1 or X.shape[1] < 2:
+        raise ValueError(
+            f"0 feature(s) (shape={X.shape}) while a minimum of 2 is required."
+        )
+
+
+def check_value_columns(X_value, **kwargs) -> np.ndarray:
+    """Do basic checks on the value columns"""
+    return check_array(X_value, **kwargs)
```

Comment:
Doesn't this function basically re-implement `check_array`?

Reply:
Yes right now it does, I put it this way for consistency in:
```
X_group = check_grouping_columns(X_group, **kwargs)
X_value = check_value_columns(X_value, **kwargs)
```
Would you prefer to have
```
X_group = check_grouping_columns(X_group, **kwargs)
X_value = check_array(X_value, **kwargs)
```
?

Reply:
The latter feels better yeah. Mainly because having another function implies that we're treating it differently.

Reply:
Alright, changed it!